File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<Integer>();
        int res = 0;
        for(int i=0;i<nums.length;i++){
            set.add(nums[i]);
        }
        for(int i=0;i<nums.length;i++){
            if(!set.contains(nums[i]-1)){
               int currSeq = 1;
               while(set.contains(nums[i]+currSeq)){
                   currSeq++;
               }
               res = Math.max(currSeq,res);
            }
        }
        return res;
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    var hash = {};
    var max = 0;
    for (var i = 0; i < nums.length; i++) {
        if (!hash[nums[i]]) {
            hash[nums[i]] = true;
            var left = nums[i] - 1;
            var right = nums[i] + 1;
            while (hash[left]) {
                left--;
            }
            while (hash[right]) {
                right++;
            }
            max = Math.max(max, right - left - 1);
        }
    }
    return max;
}